collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -644,7 +644,7 @@ private boolean loadSharedPreferencesFromFile(File src) {
                 GeneralSharedPreferences.getInstance().save(entry.getKey(), entry.getValue());
             }
 
-            AuthDialogUtility.setWebCredentialsFromPreferences();
+            new WebCredentialsUtils().setWebCredentialsFromPreferences();
```

Comment:
This suggests to me that a new WebCredentialsUtils object is being created, then mutated, and then discarded. It seems if this is meant to mutate static fields, a new object shouldn’t be created.

Reply:
I agree it does look a little odd. The reason an instance is created is because Dagger needs to dependency inject an instance of CollectServerClient into WebCredentialsUtils - Dagger can't inject into static fields.

Maybe I should create a WebCredentialsUtils factory method to return an instance of the object.  That way this detail can be hidden within WebCredentialsUtils.

Reply:
WebCredentialsUtils is now a Singleton.